src/mesh/mesh.cc

```diff
@@ -791,14 +781,14 @@ void Mesh::element_to_neigh_vb()
 
 	//MessageOut() << "Element to neighbours of vb2 type... "/*orig verb 5*/;
 
-	for (vector<Element>::iterator ele = element_vec_.begin(); ele!= element_vec_.begin()+bulk_size_; ++ele)
+	for (vector<Element>::iterator ele = element_vec_.begin(); ele!= element_vec_.end(); ++ele)
 		ele->n_neighs_vb_ =0;
 
     // count vb neighs per element
     for (auto & ngh : this->vb_neighbours_)  ngh.element()->n_neighs_vb_++;
 
     // Allocation of the array per element
-    for (vector<Element>::iterator ele = element_vec_.begin(); ele!= element_vec_.begin()+bulk_size_; ++ele)
+    for (vector<Element>::iterator ele = element_vec_.begin(); ele!= element_vec_.end(); ++ele)
```

Comment:
Same here.

Reply:
We need to iterate Element objects (not ElementAccessor) to be able to change its properties inside the loop. That is why elements_range() is not used here.